### PR TITLE
Paragraph Component Impl

### PR DESCRIPTION
**Pull request by brentmiller1973**
## Description
Ingrates the Paragraph component into the repository. 

## Jira Ticket
https://arcpublishing.atlassian.net/browse/TMEDIA-669

## Acceptance Criteria

Create a <Paragraph> component
Paragraph can be truncated - include a truncate property for number of lines with ellipses
Paragraph text can include HTML, so it should render appropriately (i.e. <b>, <i>)
No sanitization required
Have Scss file that adds the hooks to the Sass library
Have test coverage
Have storybook documentation

## Test Steps

1. Checkout this branch `TMEDIA-669-component-paragraph`
2. Run storybook `npm run storybook` and audit the stack documentation and stories
3. Run tests `npm run test` and ensure all tests run.

### CODE REVIEW

File: src/components/paragraph/index.jsx [hunk 1]
```diff
@@ -1,9 +1,33 @@
 import PropTypes from "prop-types";
 
-const Paragraph = ({ children }) => <p className="c-paragraph">{children}</p>;
+const COMPONENT_CLASS_NAME = "c-paragraph";
+
+const Paragraph = ({ children, className, truncationLines }) => (
+	<p
+		data-style-truncation-lines={truncationLines > 0 ? "true" : "false"}
+		style={{
+			WebkitLineClamp: truncationLines > 0 ? truncationLines : "none",
```
Comment (matthewroach) [inline src/components/paragraph/index.jsx]: Any reason why this is not using a CSS Variable to allow more flexibility?

Comment (brentmiller1973) [reply src/components/paragraph/index.jsx]: Doh, honestly I am going mental

Comment (brentmiller1973) [reply src/components/paragraph/index.jsx]: @matthewroach Added line clamp into the token system.  Also it's a good example cause it's a little tricky on how many dashes needed when applying a non-standard css property in React with our token system.

Comment (brentmiller1973) [reply src/components/paragraph/index.jsx]: @matthewroach Updated code based on what you did in the meeting.  Tests also modified to comply to the changes.

File: src/components/paragraph/index.jsx [hunk 2]
```diff
@@ -1,9 +1,30 @@
 import PropTypes from "prop-types";
 
-const Paragraph = ({ children }) => <p className="c-paragraph">{children}</p>;
+const COMPONENT_CLASS_NAME = "c-paragraph";
+
+const Paragraph = ({ children, className, truncationLines }) => (
+	<p
+		style={{ "--paragraph-truncation": truncationLines > 0 ? truncationLines : "none" }}
```
Comment (matthewroach) [inline src/components/paragraph/index.jsx]: The falsy part should be null - as the "none" value is already defaulted in within the Sass where you have the fallback CSS Variable declaration.
Also setting this to `null` ensure inline styles are not being set when they are not needed

```suggestion
		style={{ "--paragraph-truncation": truncationLines > 0 ? truncationLines : null }}
```

**Review by matthewroach — DISMISSED** [review]

File: src/components/paragraph/index.jsx (hunk 2 above)
Comment (brentmiller1973) [reply src/components/paragraph/index.jsx]: @matthewroach Updated code along with updated tests.

**Review by matthewroach — APPROVED** [review]

**Pull request merged**